File: test-suite/src/test/java/io/micronaut/security/docs/fetchmetadata/PublicResourceFetchMetadataRuleTest.java
package io.micronaut.security.docs.fetchmetadata;

import io.micronaut.http.HttpRequest;
import io.micronaut.security.fetchmetadata.FetchMetadataRuleResult;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class PublicResourceFetchMetadataRuleTest {

    private final PublicResourceFetchMetadataRule rule = new PublicResourceFetchMetadataRule();

    @Test
    void allowsOnlyThePublicResource() {
        assertEquals(FetchMetadataRuleResult.ALLOWED,
            rule.check(HttpRequest.GET("/favicon.ico"), null));
        assertEquals(FetchMetadataRuleResult.UNKNOWN,
            rule.check(HttpRequest.GET("/account"), null));
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/security/docs/fetchmetadata/PublicResourceFetchMetadataRuleTest.kt
package io.micronaut.security.docs.fetchmetadata

import io.micronaut.http.HttpRequest
import io.micronaut.security.fetchmetadata.FetchMetadataRuleResult
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class PublicResourceFetchMetadataRuleTest {

    private val rule = PublicResourceFetchMetadataRule()

    @Test
    fun allowsOnlyThePublicResource() {
        assertEquals(
            FetchMetadataRuleResult.ALLOWED,
            rule.check(HttpRequest.GET<Any>("/favicon.ico"), null)
        )
        assertEquals(
            FetchMetadataRuleResult.UNKNOWN,
            rule.check(HttpRequest.GET<Any>("/account"), null)
        )
    }
}
